### DISCUSSION TITLE
Welcome to OpenPineBuds Discussions!

### DISCUSSION THREAD

**Discussion by gamelaster** [pine64/OpenPineBuds#22] [Announcements]
## 👋 Welcome!
  We’re using Discussions as a place to connect with other members of our community. We hope that you:
  * Ask questions you’re wondering about.
  * Share ideas.
  * Engage with other community members.
  * Welcome others and are open-minded. Remember that this is a community we
  build together 💪.

  To get started, comment below with an introduction of yourself and tell us about what you do with this community.

**Comment by lizz-34** [comment]
Hey!

I'm Lisa, Liz for short. I have lived in the United States for four years now. I've known Pine for a long time, but I've only been involved for a while.
Last year I heard about PineBuds from my friend and got them as a gift from him (thanks to him!). In January, I got them and I'm surprised by their sound quality! In fact, one other of my friends bought them for herself after I gave them a very short test to her :) However, I see a lot of things to improvement and I'm really, really interested in open-source firmware for speakers, headphones and earbuds!

I mainly contributing to the PineBuds Pro wiki page, PineBuds Pro Matrix room and this firmware in the form of ideas. Maybe I'll also do some MRs, who knows.

-Liz

**Comment by bitflung** [comment]
I'll follow Liz's lead here.

My name is Jared, I go by 'bitflung' online. I'm a computer engineer in USA, working in semiconductors. Most of my professional and hobby work has been focused on low power embedded systems, generally using ARM Cortex M based MCUs, collecting data from sensors, and performing some locally relevant actions (e.g. opening an air vent). I bought myself a Pinecil last year and love it, which got me looking into Pine64 more closely - my wife noticed and gifted me some PineBuds over the holidays.

I'm looking forward making some contributions to the firmware. Initially I'd like to alter the default mode of operation (ambient by default, requiring taps to turn it off or enter ANC). After that i'm thinking of a firmware extension for sideband comms with an Android app to allow users to toggle ANC/Ambient settings via a notification that makes itself persistent when the buds are connected. Over time I'd like to better understand the ANC subsystem and to implement a method to allow users to modify the behavior at runtime via an android app - for example in ambient mode right now there is a slight 'hiss' sound in the background (i suspect from EMI?) which could be filtered out via compile time changes... but perhaps seeing that hiss on a spectrum plot and sliding some controls around would make for a nice intuitive method to allow users to control such behavior at runtime. That's the dream anyway, though it may be intractable in the end. It would be nice to implement a subset of those PineBud control features as an app for the PineTime too... but now i'm really dreaming big and assuming I'll have a LOT more free time than I can expect to have any day soon.

**Reply by lizz-34** [reply]
Glad you're excited about PineBuds Pro! Welcome to the [PineBuds Pro Matrix room](https://matrix.to/#/#pinebuds:matrix.org), check [this](https://github.com/pine64/OpenPineBuds/discussions/29) and [this](https://github.com/pine64/OpenPineBuds/issues/31) too.

>  Initially I'd like to alter the default mode of operation (ambient by default, requiring taps to turn it off or enter ANC). After that i'm thinking of a firmware extension for sideband comms with an Android app to allow users to toggle ANC/Ambient settings via a notification that makes itself persistent when the buds are connected.

This firmware doesn't yet support ANC at all, but it's work in progress. The firmware currently pre-installed in PineBuds Pro is proprietary, closed source.

**Comment by shymega** [comment]
Hey all.

I reached out to Pine last year about helping develop the PineBuds, and I'm glad to announce that I've been added as a collaborator to this repository. 

I should be receiving a pair of PineBuds soon, and I hope that a PineSound is available too. I've been working with the team for a while, and some of you may have seen my PRs to the firmware, and talk on the PineBuds channel.

I'm excited to be able to contribute to this project as a developer, and I've got a branch on this repository to convert over to CMake.

Future ideas I have in mind, as well as what's been discussed here on GitHub, are as follows, in no particular order:

- Conversions of Makefiles to CMake, so we can support IDEs better, and Ninja.
- Cataloguing of licenses on files, including blobs, using FSFE's 'REUSE' utility.
- Adding a Code of Conduct.
- Support for Bluetooth Multipoint!
- Tracking down external contributors, and potentially using a CLA to resolve copyright issues.
- Labelling issues.
- Slowly rewriting the C/C++/ASM codebase into Rust. We depend on CMake for this to be made possible. The benefit of this is that we can use a memory-safe RTOS.
- Dropping unnecessary 'apps' and 'services' from the SDK.
- Fixing bug reports.
- And finally, adding BLE/RFCOMM support for GATT states, and control of the buds. This will allow for third-party apps to interface with the buds!

It's a lot. Help is welcome, of course.

Thank you!
-
